pytorch_lightning/callbacks/prediction_writer.py

```diff
@@ -0,0 +1,99 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+from typing import Any, List, Optional
+
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class PredictionWriterBase(Callback):
+
+    write_intervals = ("batch", "epoch", "batch_and_epoch")
+    """
```

Comment:
Maybe this is nonsense but I think the docstring should be right at the top

Sphinx might have issues with having a class attribute before

Reply:
Done !